practice/prototype_anc_samp.py

```diff
@@ -0,0 +1,624 @@
+# Advanced prototype:
+# Does GC at user-specified intervals
+# Random number of edges due to modeling
+# recombination as a Poisson process.
+
+import numpy as np
+import msprime
+import sys
+import argparse
+import pickle
+from collections import namedtuple
+
+node_dt = np.dtype([('id', np.uint32),
+                    ('generation', np.float),
+                    ('population', np.int32)])
+
+edge_dt = np.dtype([('left', np.float),
+                    ('right', np.float),
+                    ('parent', np.int32),
+                    ('child', np.int32)])
+
+mutation_dt = np.dtype([('position', np.float64),
+                        ('node_id', np.int32),
+                        ('origin_generation', np.float64)])
+
+# Simulation with be popsize*SIMLEN generations
+SIMLEN = 20
+
+
+class MockAncestryTracker(object):
+    """
+    Mimicking the public API of AncestryTracker.
+    """
+    __nodes = None
+    __edges = None
+    __mutations = None
+    __samples = None
+    __anc_samples = None
+
+    def __init__(self):
+        self.nodes = np.empty([0], dtype=node_dt)
+        self.edges = np.empty([0], dtype=edge_dt)
+        self.mutations = np.empty([0], dtype=mutation_dt)
+        self.samples = None
+        self.anc_samples = []
```

Comment:
This should probably be `np.empty`, as you use it as a numpy array later on.

Reply:
thanks! just pushed a commit to fix this